### PR TITLE
using LINQ in the code

### PR DESCRIPTION
**Pull request by Jankowski-J**
Masz tu @Rychu-Pawel trochę LINQ.

### CODE REVIEW

File: src/PeselValidator/PeselValidator.cs [hunk 1]
```diff
@@ -45,15 +45,12 @@ private static bool IsChecksumValid(string pesel)
...
+            return pesel.Last().ToString() == checksum.ToString();
         }
 
         private static int CalculateChecksum(string pesel)
         {
-            int sum = 0;
-
-            for (int i = 0; i < 10; i++)
-                sum += weights[i] * int.Parse(pesel[i].ToString());
+            var sum = weights.Select((x, i) => x * int.Parse(pesel[i].ToString())).Sum();
```
Comment (Rychu-Pawel) [inline src/PeselValidator/PeselValidator.cs]: Thanks @Jankowski-J for this PR. I checked the performance of this code and unfortunatly it is 3 times slower than the original one. 
I like the change with `Last()` though. It clrealy highlights the intent.
I did some more testing and looks like we can get the lost performance back by using `Convert.ToInt32` in line 53 and `Convert.ToChar` in line 48. Then these three `.ToString()`s are redundant and can be removed.
Could you please implement these changes?

**Review by Rychu-Pawel — CHANGES_REQUESTED** [review]
Thanks @Jankowski-J for this PR. I checked the performance of this code and unfortunatly it is 3 times slower than the original one.
I like the change with `Last()` though. It clrealy highlights the intent.
I did some more testing and looks like we can get the lost performance back by using `Convert.ToInt32` in line 53 and `Convert.ToChar` in line 48. Then these three `.ToString()`s are redundant and can be removed.
Could you please implement these changes?

**Comment by Rychu-Pawel** [comment]
Would it be also possible to rebase your branch on the latest develop? It should fix appveyor build crashing on sonarqube analysis. Thanks in advance.

**Comment by Jankowski-J** [comment]
Suggested fixes - using `Convert` methods - fail the tests.
Using `Convert` would still require using `.ToString()` to produce valid output.
Since you are concerned about the performance, should this PR be abandoned?

**Comment by Rychu-Pawel** [comment]
@Jankowski-J 
The biggest performance killer is the change in `CalculateChecksum`. Here I would stick to the standard `for` loop. The change with `.Last()`, on the other hand, looks fine for me. I would like to merge this one.
And, yea, of course `Convert` breaks the tests. My bad. Sorry. Could you please use `char.GetNumericValue` instead? This one works for sure and is faster than comparing `.ToString`s :)

**Comment by Jankowski-J** [comment]
Changed as per suggestions.

**Review by Rychu-Pawel — APPROVED** [review]

**Comment by Jankowski-J** [comment]
:partyparrot:

Pozdrawiam,
Jakub Jankowski

________________________________

**Pull request merged**